Completely remove AGB Stekene's data

Riadabd opened the pull request:
## Ticket ID

OP-3409

## PR Description

A previous PR only updated the org's status to `Inactive`. However, we should now delete all data related to it as Loket doesn't have a notion of `status`, which can lead to issues.

## How to Test

Tunnel to the database and go over the forward and reverse connections to this organization. After running the migration, all data should be deleted.

> Note: I left the comments in the migration for the purpose of separating the components and making it easier to have a look at what is going to be deleted. The comments will be removed once the PR is approved since `mu-auth` doesn't accept them.

mirdono reviewed (changes requested):
Some comments:

**Issue:** This original archive migration has already been executed on DEV, so you will have to put this revised version in a new file for it to be executed there.

**Issue:** This organisation also has a site which is currently not removed:`<http://data.lblod.info/id/vestigingen/daa79855-1de7-4479-8a2e-a8d89db55ee3>`

**Issue:** On DEV this organisations is also involved in three memberships which should also be removed:

| subject                                                                         | predicate                             | object                                                                          |
|--------------------------------------------------------------------------------- | --------------------------------------- | ---------------------------------------------------------------------------------|
| http://data.lblod.info/id/lidmaatschap/1c2608c9cde34c3091442af5df63aa7cb27e7022 | http://www.w3.org/ns/org#organization | http://data.lblod.info/id/bestuurseenheden/8d6609b4-1a20-46b8-97ed-150c6d53acd6 |
| http://data.lblod.info/id/lidmaatschap/958418b8cd7d5c5668ec4629d0846cba33eaf761 | http://www.w3.org/ns/org#organization | http://data.lblod.info/id/bestuurseenheden/8d6609b4-1a20-46b8-97ed-150c6d53acd6 |
| http://data.lblod.info/id/lidmaatschap/92316facb819628fd41ed8dd18ba9ec902383e54 | http://www.w3.org/ns/org#member       | http://data.lblod.info/id/bestuurseenheden/8d6609b4-1a20-46b8-97ed-150c6d53acd6 |

Riadabd commented:
> Issue: This original archive migration has already been executed on DEV, so you will have to put this revised version in a new file for it to be executed there.

Ah, thanks for letting me know :) I've restored the old migration and made a new one.

> Issue: This organisation also has a site which is currently not removed:<http://data.lblod.info/id/vestigingen/daa79855-1de7-4479-8a2e-a8d89db55ee3>

Oh, I missed that one. Nice catch!

> Issue: On DEV this organisations is also involved in three memberships which should also be removed.

Done.

mirdono commented on config/migrations-triggering-indexing/2025/20250109200145-archive-agb-stekene.sparql:
```diff
@@ -0,0 +1,200 @@
+
+  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
+    ?parentOrganization <http://www.w3.org/ns/org#hasSubOrganization> ?organization ;
+      <http://mu.semte.ch/vocabularies/ext/isFounderOfOrganization> ?organization .
```
**Issue:** On DEV this `ext:isFounderOfOrganization` triple does not exist anymore (it was replaced by a membership), causing this query to do nothing on that environment. On the other environments any `ext:isFounderOfOrganization` triples should be deleted as well as part of introducing memberships (cf. [the migration](https://github.com/lblod/app-organization-portal/blob/34d8ec58183bd9f1b08a2de0c92a37d067f8fe63/config/migrations/2024/20240704112300-related-organizations-as-memberships/20240704112600-founding-members.sparql)). So I think specifying the founder triple is not needed at all.

mirdono reviewed (changes requested):
**Suggestion:** I noticed that on DEV there is also data in the worship graph for this organisation's URI for some unknown reason. Would you mind also removing that as well? Either in this migration or using another local migration on DEV.

```sparql
SELECT DISTINCT ?p ?o
WHERE {
  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    <http://data.lblod.info/id/bestuurseenheden/8d6609b4-1a20-46b8-97ed-150c6d53acd6> ?p ?o.
  }
}
```

| p                                               | o                                                                          |
|-------------------------------------------------|----------------------------------------------------------------------------|
| http://www.w3.org/1999/02/22-rdf-syntax-ns#type | http://data.vlaanderen.be/ns/besluit#Bestuurseenheid                       |
| http://mu.semte.ch/vocabularies/core/uuid       | 8d6609b4-1a20-46b8-97ed-150c6d53acd6                                       |
| http://www.w3.org/ns/org#hasPrimarySite         | http://data.lblod.info/id/vestigingen/daa79855-1de7-4479-8a2e-a8d89db55ee3 |

Riadabd commented:
> **Suggestion:** I noticed that on DEV there is also data in the worship graph for this organisation's URI for some unknown reason. Would you mind also removing that as well? Either in this migration or using another local migration on DEV.

I removed the memberships from this PR's migration and will instead bundle them as part of a local migration to be run on DEV only. I will also include deleting the data belonging to `AGB Stekene` that are present in the `<http://mu.semte.ch/graphs/worship-service>` graph on DEV.

<details>
<summary>This is the migration that will run on DEV.</summary>

```sparql
DELETE {
  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
    ?membershipOrganization <http://www.w3.org/ns/org#organization> ?membershipOrganizationO .
  }
  
  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
    ?membershipMember <http://www.w3.org/ns/org#member> ?membershipMemberO .
  }
}
WHERE {
  VALUES ?membershipOrganization {
    <http://data.lblod.info/id/lidmaatschap/1c2608c9cde34c3091442af5df63aa7cb27e7022>
    <http://data.lblod.info/id/lidmaatschap/958418b8cd7d5c5668ec4629d0846cba33eaf761>
  }
  
  VALUES ?membershipMember {
    <http://data.lblod.info/id/lidmaatschap/92316facb819628fd41ed8dd18ba9ec902383e54>
  }

  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
    ?membershipOrganization <http://www.w3.org/ns/org#organization> ?membershipOrganizationO .
  }
  
  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
    ?membershipMember <http://www.w3.org/ns/org#member> ?membershipMemberO .
  }
}

;

DELETE {
  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?organization ?p ?o .
  }
}
WHERE {
  BIND(<http://data.lblod.info/id/bestuurseenheden/8d6609b4-1a20-46b8-97ed-150c6d53acd6> AS ?organization)

  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?organization ?p ?o .
  }
}

;

DELETE {
  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?site ?siteP ?siteO .
  }
}
WHERE {
  VALUES ?site {
    <http://data.lblod.info/id/vestigingen/daa79855-1de7-4479-8a2e-a8d89db55ee3>
  }

  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?site ?siteP ?siteO .
  }
}

;

DELETE {
  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?address ?addressP ?addressO .
  }
}
WHERE {
  VALUES ?address {
    <http://data.lblod.info/id/adressen/df102cb4-8abb-4588-a199-1ce22c211812>
  }

  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?address ?addressP ?addressO .
  }
}

;

DELETE {
  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?contactPoint ?contactPointP ?contactPointO .
  }
}
WHERE {
  VALUES ?contactPoint {
    <http://data.lblod.info/id/contact-punten/042e8816-2e9d-45a1-89f6-dcf805882af7>
    <http://data.lblod.info/id/contact-punten/364d8e1a-cb18-4970-aeaf-5f6f307370ad>
  }

  GRAPH <http://mu.semte.ch/graphs/worship-service> {
    ?contactPoint ?contactPointP ?contactPointO .
  }
}
```

</details>

mirdono reviewed (commented):
I would keep the membership deletion as part of this PR. The migrations to convert existing relations to memberships will be executed before this migration when everything in `development` finally moves to a (pre-)release on QA/PROD. So the new created memberships for this organisation would have to be immediately deleted there.

Sorry, the significant discrepancy between DEV and QA/PROD is making this deletion way harder than it should be.

For deleting the memberships, I would generalize it to something like this (not thoroughly tested):
```sparql
PREFIX org: <http://www.w3.org/ns/org#>

DELETE {
  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
    ?membership ?p ?o.
  }
}
WHERE {
  GRAPH <http://mu.semte.ch/graphs/administrative-unit> {
    ?membership a org:Membership;
                ?membershipP <http://data.lblod.info/id/bestuurseenheden/8d6609b4-1a20-46b8-97ed-150c6d53acd6>;
                ?p ?o.
  }
  VALUES ?membershipP {
    org:organization
    org:member
  }
}
```

Note: the migrations creating the memberships do generate the membership UUIDs based on the URIs of the involved organisations. So technically the membership URIs should be the same across all environments assuming that the URIs for the organisation are consistent. But I did not explicitly check that for this case and, imo, it is easier/more reliable to use a more general query as above.

Riadabd commented:
I updated the query in my earlier comment, and the local dev migration will only be responsible for removing the org's data from the worship graph.

I also updated the PR migration to remove the membership triples, as you mentioned.

mirdono reviewed (approved):
Looks good, the migrations covers all data now and the DEV specific works as well.

One final detail, would you mind squashing the commits a bit before merging, imo the intermediate changes are not needed in the main history.
